trustymail/trustymail.py

```diff
@@ -63,6 +70,99 @@ def mx_scan(domain):
         handle_error("[MX]", domain, error)
 
 
+def starttls_scan(domain, smtp_timeout, smtp_localhost, smtp_ports, smtp_cache):
+    """
+    Scan a domain to see if it supports SMTP and supports STARTTLS.
+
+    Scan a domain to see if it supports SMTP.  If the domain does support
+    SMTP, a further check will be done to see if it supports STARTTLS.
+    All results are stored inside the Domain object that is passed in
+    as a parameter.
+
+    Parameters
+    ----------
+    domain : Domain
+        The Domain to be tested.
+        
+    smtp_timeout : int
+        The SMTP connection timeout in seconds.
+
+    smtp_localhost : str
+        The hostname to use when connecting to SMTP servers.
+
+    smtp_ports : obj:`list` of :obj:`str`
+        A comma-delimited list of ports at which to look for SMTP servers.  
+        
+    smtp_cache : bool
+        Whether or not to cache SMTP results.
+    """
+    for mail_server in domain.mail_servers:
+        for port in smtp_ports:
+            if port not in domain.ports_tested:
+                domain.ports_tested.append(port)
```

Comment:
I wonder if it would be clearer to have `domain.ports_tested` be a `set` rather than a `list` and then you could just do: `domain.ports_tested.add(port)` and you don't need to do the conditional check of if it's in there.